### ISSUE TITLE
Some debug flags are not producing any output.

### ISSUE THREAD

**Issue by chevathamanon** [OpenModelica/OpenModelica#11798]
**Problem:**
Translational flags are not producing any output in the log file.

**Tested version:**
OMEdit v. 1.20 and 1.22.1 (windows)

**Model:**
any model.

**Flags:**
+d=bltdump
+d=backenddaeinfo
-d=bltdump
-d=backenddaeinfo
and so on..

Would it be able to fix this?

**Comment by chevathamanon** [comment]
Dear Prof. Casella,
Thank you very much for concerning this issue. To be more specific, I created a simple model "SpringMass" to test thetranslational flags. For this, I expected to see additional background information once the simulation runs.

At first, In order to make sure that all settings are kept default, Openmodelica 1.22.1 (64bits) windows has been reinstalled. By using the following model "SYS_SingleSpringMass" (attachment), an additional translational flags "+d=bltdump" has been added to the setting (please see the attached screenshot). Afterwards, run the simulation. No further background information has been printed under the output tab (or any other tab) also no additional files containing background information have been created under the working folder.

Basically, I have tested not only the flag "bltdump" but also other flags. Nothing has been printed as well. 

Note: the only flag that seems to work but was "bltmatrixdump", which created Html files. However, without any information printed under the log file.

The flags that I have tried are from
https://openmodelica.org/doc/OpenModelicaUsersGuide/1.22/omchelptext.html#omcflag-debug-section

Attachment: Model + Screenshot
[SpringMass.zip](https://github.com/OpenModelica/OpenModelica/files/13890727/SpringMass.zip)

Thank you very much,

Sincerely yours,

Chevathamanon.

**Comment by casella** [comment]
@chevathamanon thanks for reporting this; unfortunately we have no means to automatically test all these GUI aspects, so mayb something went wrong with the latest releases. I'll investigate and get back to you ASAP

**Comment by perost** [comment]
> OK, I made some more tests with a script like this one
> 
> ```
> loadModel(Modelica);getErrorString();
> setCommandLineOptions("-d=bltdump");getErrorString();
> simulate(Modelica.Blocks.Examples.Filter);getErrorString();
> ```
> 
> to check if this is an OMEdit problem or an OMC problem. Apparently, some debug flags do not produce any output, e.g.: -d=bltdump, -d=bltmatrixdump, while other do, e.g. -d=execstat or -d=optdaedump.
> 
> @phannebohm, @adrpo, can you please investigate?

It seems like `Modelica.Blocks.Examples.Filter` simply doesn't hit any of the places in the compiler where `bltdump` is used, if you try e.g. `Modelica.Fluid.Eamples.HeatingSystem` instead you'll get some output from that flag.

**Comment by chevathamanon** [comment]
@perost Thank you very much for helping to investigate the problem. Would there be any way to check the necessary conditions for the flags? So that we can expect hit or miss.

**Comment by casella** [comment]
I guess -d=bltdump is useful when you have strong components in the BLT, which is not the case for models like spring-mass or filters that can be sorted out into a sequence of explicit assignments.

I'll ask for a confirmation at today's devemeeting.

If you are interested in the structure of the solved model, please use the [Transformational Debugger](https://openmodelica.org/doc/OpenModelicaUsersGuide/latest/debugger.html#debugging).

**Comment by casella** [comment]
OK, here is some more info for you
- The transformation debugger gets all its information from the `Modelname_info.json` file that is generated together with the executable code. If you are using OMEdit, you find it in the subdirectory of the working directory which is named as the fully qualified name of the model you are compiling and simulating. You can parse that json file and get the same information that is displayed in the debugger.
- `-d=bltdump` only outputs information when index reduction is required - I'll see if I can update the flag description
- `-d=bltmatrixdump` outputs the structure of the BLT matrix graphically as an HTML file - the cross-references to the equations and variables can be found in the json file

**Comment by chevathamanon** [comment]
Thank you very much.
